outofcore/src/cJSON.cpp

```diff
@@ -496,7 +499,7 @@ cJSON *cJSON_GetObjectItem(cJSON *object,const char *string)	{cJSON *c=object->c
 /* Utility for array list handling. */
 static void suffix_object(cJSON *prev,cJSON *item) {prev->next=item;item->prev=prev;}
 /* Utility for handling references. */
-static cJSON *create_reference(cJSON *item) {cJSON *ref=cJSON_New_Item();if (!ref) return nullptr;memcpy(ref,item,sizeof(cJSON));ref->string=nullptr;ref->type|=cJSON_IsReference;ref->next=ref->prev=nullptr;return ref;}
+static cJSON *create_reference(cJSON *item) {cJSON *ref=cJSON_New_Item();if (!ref) return nullptr;std::copy_n(item, sizeof(cJSON), ref);ref->string=nullptr;ref->type|=cJSON_IsReference;ref->next=ref->prev=nullptr;return ref;}
```

Comment:
Please apply some basic formatting on this line. Reviewing the change is a bit hard.

Also, I feel `copy_n` is bad choice since `n` might just be 1 based on the old code

Reply:
Reformatted. `sizeof(cJSON)` > 1, even in the old code, isn't it?

Reply:
`copy` works on objects, while `memcpy` works on bytes. That means 5 byte object needs 5 as argument for memcpy but 1 for copy

Reply:
(facepalm)